net-misc/chrony/chrony-4.0_pre1-r2.ebuild

```diff
@@ -0,0 +1,183 @@
+	fi
+
+	if ! use seccomp; then
+		sed -i \
+			-e 's/-F 1//' \
+			"${T}"/chronyd.conf "${T}"/chronyd.service || die
+	fi
+}
+
+src_configure() {
+	tc-export CC
+
+	local CHRONY_EDITLINE
+	# ./configure legend:
+	# --disable-readline : disable line editing entirely
+	# --without-readline : do not use sys-libs/readline (enabled by default)
+	# --without-editline : do not use dev-libs/libedit (enabled by default)
+	if ! use readline && ! use libedit; then
+		CHRONY_EDITLINE='--disable-readline'
+	else
+		CHRONY_EDITLINE+=" $(usex readline '' --without-readline)"
+		CHRONY_EDITLINE+=" $(usex libedit '' --without-editline)"
+	fi
+
+	# not an autotools generated script
+	local myconf=(
+		$(use_enable seccomp scfilter)
+		$(usex adns '' --disable-asyncdns)
+		$(usex caps '' --disable-linuxcaps)
+		$(usex cmdmon '' --disable-cmdmon)
+		$(usex ipv6 '' --disable-ipv6)
+		$(usex nettle '' --without-nettle)
+		$(usex ntp '' --disable-ntp)
+		$(usex phc '' --disable-phc)
+		$(usex pps '' --disable-pps)
+		$(usex refclock '' --disable-refclock)
+		$(usex rtc '' --disable-rtc)
+		$(usex sechash '' --disable-sechash)
+		${CHRONY_EDITLINE}
+		${EXTRA_ECONF}
+		--chronysockdir="${EPREFIX}/run/chrony"
+		--docdir="${EPREFIX}/usr/share/doc/${PF}"
+		--mandir="${EPREFIX}/usr/share/man"
+		--prefix="${EPREFIX}/usr"
+		--sysconfdir="${EPREFIX}/etc/chrony"
+		--with-pidfile="${EPREFIX}/run/chrony/chronyd.pid"
+		--without-nss
+		--without-tomcrypt
+	)
+
+	# print the ./configure call to aid in future debugging
+	echo bash ./configure "${myconf[@]}" >&2
+	bash ./configure "${myconf[@]}" || die
+}
+
+src_compile() {
+	emake all docs $(usex html '' 'ADOC=true')
+}
+
+src_install() {
+	default
+
+	newinitd "${FILESDIR}"/chronyd.init-r2 chronyd
+	newconfd "${T}"/chronyd.conf chronyd
+
+	insinto /etc/${PN}
+	newins examples/chrony.conf.example1 chrony.conf
+
+	docinto examples
+	dodoc examples/*.example*
+
+	newtmpfiles - chronyd.conf <<<"d /run/chrony 0750 $(usex caps 'ntp ntp' 'root root')"
+
+	if use html; then
+		docinto html
+		dodoc doc/*.html
+	fi
+
+	if use caps; then
+		# Prepare a directory for the chrony.drift file (a la ntpsec)
+		# Ensures the environment is sane on new installs
+		mkdir -pv "${ED}"/var/lib/chrony || die
+		fowners ntp:ntp /var/lib/chrony
+		fperms 770 /var/lib/chrony
+	fi
+
+	keepdir /var/{lib,log}/chrony
+
+	insinto /etc/logrotate.d
+	newins "${FILESDIR}"/chrony-2.4-r1.logrotate chrony
+
+	systemd_dounit "${T}"/chronyd.service
+	systemd_dounit examples/chrony-wait.service
+	systemd_enable_ntpunit 50-chrony chronyd.service
+}
+
+pkg_postinst() {
+	tmpfiles_process chronyd.conf
+
+	if use caps && ! use prefix; then
+		# Permission changes on the livefs are not ideal.
+		# Let's make sure they're necessary.
+		# Versions with incorrect perms: <=chrony-3.5-r3, <=4.0_pre1-r1
+		if [[ -n ${REPLACING_VERSIONS} ]]; then
+			for ver in ${REPLACING_VERSIONS}; do
+				if ver_test ${ver} -le 3.5-r3 || ver_test ${ver} -le 4.0_pre1-r1; then
```

Comment:
This is not doing what you expect. I.e. people with 3.5-r4 (or 3.6 in case there will be such a release) would still get the message because `ver_test ${ver} -le 4.0_pre1-r1` is true for them.